### PR TITLE
Rewrite EntryPointManager as Popover subclass

### LINKED ISSUE #444: Refactor popover instantiation
The entry point popover manager should not be a manager, it should just be a popover. Similar to how the PreferencesPopover is instantiated.

### PR DESCRIPTION
**Pull request by maxmmyron**
Fixes #444

This PR refactors the `EntryPointManager` class to a subclass of `Popover`.

This PR also fixes a small issue where the curtain would not appear, leading to some potential UI issues, and moves one test case to a path that better mirrors the existing `studio/` file structure.

### CODE REVIEW

File: studio/src/windowManagement/contentWindows/ContentWindowBuildView/ContentWindowBuildView.js [hunk 1]
```diff
@@ -82,15 +82,15 @@ export class ContentWindowBuildView extends ContentWindow {
...
-				const studio = getStudioInstance();
-				const projectSettings = studio.projectManager.projectSettings;
-				const assetManager = studio.projectManager.assetManager;
-				if (!projectSettings || !assetManager) return;
-				const popover = studio.popoverManager.createPopover();
-				// eslint-disable-next-line no-new
-				new EntryPointManager(popover, projectSettings, assetManager, this.persistentData);
-
-				popover.setNeedsCurtain(false);
+				const projectSettings = this.studioInstance.projectManager.projectSettings;
```
Comment (jespertheend) [inline studio/src/windowManagement/contentWindows/ContentWindowBuildView/ContentWindowBuildView.js:85]: Nice cleanup! There's a few more calls to `getStudioInstance()` in this file. Feel free to replace those as well, then the `getStudioInstance` import can be completely removed.

File: studio/src/windowManagement/contentWindows/ContentWindowBuildView/ContentWindowBuildView.js [hunk 2]
```diff
@@ -82,15 +82,15 @@ export class ContentWindowBuildView extends ContentWindow {
...
 			onClick: () => {
-				const studio = getStudioInstance();
-				const projectSettings = studio.projectManager.projectSettings;
-				const assetManager = studio.projectManager.assetManager;
-				if (!projectSettings || !assetManager) return;
-				const popover = studio.popoverManager.createPopover();
-				// eslint-disable-next-line no-new
-				new EntryPointManager(popover, projectSettings, assetManager, this.persistentData);
-
-				popover.setNeedsCurtain(false);
```
Comment (jespertheend) [inline studio/src/windowManagement/contentWindows/ContentWindowBuildView/ContentWindowBuildView.js:93]: The curtain was disabled to allow you to drag assets from the project window to the entry point dropdown gui.
Without this the popover will close when you try to drag assets.
Ideally a test should have caught this, so I've opened up #504

File: studio/src/windowManagement/contentWindows/ContentWindowBuildView/EntryPointPopover.js [hunk 3]
```diff
@@ -38,34 +39,37 @@ export async function getSelectedEntryPoint(projectSettingsManager, contentWindo
...
-	constructor(popover, projectSettingsManager, assetManager, persistentData) {
-		this.#projectSettings = projectSettingsManager;
-		this.#assetManager = assetManager;
-		this.#persistentData = persistentData;
+	constructor(...args) {
+		super(...args);
 
 		this.#selectorContainer = document.createElement("div");
-		popover.el.appendChild(this.#selectorContainer);
+		// TODO: double check "this" works
```
Comment (jespertheend) [inline studio/src/windowManagement/contentWindows/ContentWindowBuildView/EntryPointPopover.js:65]: I'm not sure what this `TODO` means, I tried it out on https://pr-502.renda.studio/ and it seems to be working.
Is there a circumstance that hasn't been tested yet?

**Review by jespertheend — CHANGES_REQUESTED** [review]
Thanks, there's a few minor issues, but looking good other than that!

File: studio/src/windowManagement/contentWindows/ContentWindowBuildView/EntryPointPopover.js (hunk 3 above)
Comment (maxmmyron) [reply studio/src/windowManagement/contentWindows/ContentWindowBuildView/EntryPointPopover.js:65]: Whoops, that was just a mental note to double check the `Popover` class properties. Forgot to remove it!

**Pull request merged**